File: algorithms/1_easy/U/03978_unique_middle_element/solution.c
bool isMiddleElementUnique(int* nums, const int len) {
    assert(nums != NULL && len > 0 && (len & 1) == 1);
    const int mid_idx = len / 2;
    const int mid_val = nums[mid_idx];

    for (int i = 0; i < mid_idx; i += 1) {
        if (nums[i] == mid_val || nums[i + mid_idx + 1] == mid_val) {
            return false;
        }
    }

    return true;
}

File: algorithms/1_easy/U/03978_unique_middle_element/solution.rs
impl Solution {
    pub fn is_middle_element_unique(nums: Vec<i32>) -> bool {
        assert!(nums.len() > 0 && (nums.len() & 1) == 1);
        let mid_idx = nums.len() / 2;
        let mid_val = nums[mid_idx];

        for i in 0..mid_idx {
            if nums[i] == mid_val || nums[i + mid_idx + 1] == mid_val {
                return false;
            }
        }

        true
    }
}
